src/TheThingsNetwork.cpp

```diff
@@ -436,6 +443,17 @@ void TheThingsNetwork::reset(bool adr)
     sendMacSet(MAC_ADR, "off");
   }
   this->adr = adr;
+  this->needsHardReset = false;
+}
+
+void TheThingsNetwork::resetHard(uint8_t resetPin){
+  digitalWrite(resetPin, LOW);
+  unsigned long time_now = milis();
+  int period = 1000;
+  while(millis() < time_now + period){
+    // wait
```

Comment:
It was just to make it non-blocking in case the application is using interrupt. The reset function is not time critical. 
I have looked in the datasheet and there is no advice on minimum time for the signal for a reset to be triggered. I know that 1s works by test, but I have not tested <1s.

Reply:
Actually delay() is the better one to use. On some platforms where it is necessary delay() will call yield(), while a while loop will not.

Reply:
Can you implement this @savnik ?